### ISSUE TITLE
The `KnownLayout` macro has an issue with the latest nightly

### ISSUE THREAD

**Issue by AaronKutch** [google/zerocopy#2117]
This is failing for me with `nightly-2024-11-26`
```
use zerocopy::{FromBytes, Immutable, IntoBytes, KnownLayout, Unaligned};

#[derive(
    Clone, Copy, Hash, PartialEq, Eq, FromBytes, IntoBytes, KnownLayout, Immutable, Unaligned,
)]
#[repr(C)]
pub struct Test(pub [u8; 32]);

```
running `cargo +nightly-2024-11-26 clippy` gives
```
warning: unexpected `cfg` condition name: `coverage_nightly`
 --> src/main.rs:5:61
  |
5 |     Clone, Copy, Hash, PartialEq, Eq, FromBytes, IntoBytes, KnownLayout, Immutable, Unaligned,
  |                                                             ^^^^^^^^^^^
  |
  = help: expected names are: `clippy`, `debug_assertions`, `doc`, `docsrs`, `doctest`, `feature`, `fmt_debug`, `miri`, `overflow_checks`, `panic`, `proc_macro`, `relocation_model`, `rustfmt`, `sanitize`, `sanitizer_cfi_generalize_pointers`, `sanitizer_cfi_normalize_integers`, `target_abi`, `target_arch`, `target_endian`, `target_env`, `target_family`, `target_feature`, `target_has_atomic`, `target_has_atomic_equal_alignment`, `target_has_atomic_load_store`, `target_os`, `target_pointer_width`, `target_thread_local`, `target_vendor`, `test`, `ub_checks`, `unix`, and `windows`
  = help: consider using a Cargo feature instead
  = help: or consider adding in `Cargo.toml` the `check-cfg` lint config for the lint:
           [lints.rust]
           unexpected_cfgs = { level = "warn", check-cfg = ['cfg(coverage_nightly)'] }
  = help: or consider adding `println!("cargo::rustc-check-cfg=cfg(coverage_nightly)");` to the top of the `build.rs`
  = note: see <https://doc.rust-lang.org/nightly/rustc/check-cfg/cargo-specifics.html> for more information about checking conditional configuration
  = note: `#[warn(unexpected_cfgs)]` on by default
  = note: this warning originates in the derive macro `KnownLayout` (in Nightly builds, run with -Z macro-backtrace for more info)
```
which is causing issues in my strict CI

**Comment by joshlf** [comment]
I am able to reproduce this locally using zerocopy 0.8.11. Confusingly, when I run `cargo expand`, I don't see the string `coverage_nightly` anywhere in the output. That makes me think that this might be a compiler bug.

What was the most recent nightly for which this code passed Clippy without issue?

**Comment by joshlf** [comment]
I've filed an issue upstream: https://github.com/rust-lang/rust/issues/133775

**Comment by joshlf** [comment]
Update: [It's actually a zerocopy bug](https://github.com/rust-lang/rust/issues/133775#issuecomment-2514903385).

Fix pending: https://github.com/google/zerocopy/pull/2123

**Comment by joshlf** [comment]
Fixed by #2123.

**Comment by joshlf** [comment]
The fix is now released in [0.8.12](https://crates.io/crates/zerocopy/0.8.12).
